### PR TITLE
feat: increase max values for lf tags

### PR DESCRIPTION
**Pull request by frco9**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### Feature details
When using the AWS console to create a LF Tag and its values, we can only define 15 values. However when using the `aws cli` and the API, the maximum values allowed per call is 50. Furthermore, it's indicated in this [documentation](https://docs.aws.amazon.com/lake-formation/latest/dg/TBAC-notes.html) that the soft-limit for values per tag is 1000. 

This PR allows user to create tag values up to this limit. 

<ins>Output from acceptance testing:</ins>

```
$ make testacc TESTS=TestAccLakeFormation_serial/LFTags PKG=lakeformation
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lakeformation/... -v -count 1 -parallel 20 -run='TestAccLakeFormation_serial/LFTags'  -timeout 180m
=== RUN   TestAccLakeFormation_serial
=== RUN   TestAccLakeFormation_serial/LFTags
=== RUN   TestAccLakeFormation_serial/LFTags/basic
=== RUN   TestAccLakeFormation_serial/LFTags/many_values
=== RUN   TestAccLakeFormation_serial/LFTags/disappears
=== RUN   TestAccLakeFormation_serial/LFTags/values
=== RUN   TestAccLakeFormation_serial/ResourceLFTags
=== RUN   TestAccLakeFormation_serial/ResourceLFTags/tableWithColumns
=== RUN   TestAccLakeFormation_serial/ResourceLFTags/basic
=== RUN   TestAccLakeFormation_serial/ResourceLFTags/database
=== RUN   TestAccLakeFormation_serial/ResourceLFTags/databaseMultiple
=== RUN   TestAccLakeFormation_serial/ResourceLFTags/table
--- PASS: TestAccLakeFormation_serial (134.03s)
    --- PASS: TestAccLakeFormation_serial/LFTags (51.81s)
        --- PASS: TestAccLakeFormation_serial/LFTags/basic (9.82s)
        --- PASS: TestAccLakeFormation_serial/LFTags/many_values (9.46s)
        --- PASS: TestAccLakeFormation_serial/LFTags/disappears (8.06s)
        --- PASS: TestAccLakeFormation_serial/LFTags/values (24.46s)
    --- PASS: TestAccLakeFormation_serial/ResourceLFTags (82.22s)
        --- PASS: TestAccLakeFormation_serial/ResourceLFTags/tableWithColumns (19.14s)
        --- PASS: TestAccLakeFormation_serial/ResourceLFTags/basic (8.98s)
        --- PASS: TestAccLakeFormation_serial/ResourceLFTags/database (17.03s)
        --- PASS: TestAccLakeFormation_serial/ResourceLFTags/databaseMultiple (18.12s)
        --- PASS: TestAccLakeFormation_serial/ResourceLFTags/table (18.95s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lakeformation	135.724s

```

### CODE REVIEW

File: internal/service/lakeformation/lf_tag.go [hunk 1]
```diff
@@ -42,7 +45,7 @@ func ResourceLFTag() *schema.Resource {
 				Type:     schema.TypeSet,
 				Required: true,
 				MinItems: 1,
-				MaxItems: 15,
+				MaxItems: 1000,
```
Comment (theophilechevalier) [inline internal/service/lakeformation/lf_tag.go:54]: Maybe add a comment stating that it is the AWS limit?

File: internal/service/lakeformation/lf_tag.go [hunk 2]
```diff
@@ -66,17 +70,45 @@ func resourceLFTagCreate(d *schema.ResourceData, meta interface{}) error {
...
+		TagValues: flex.ExpandStringSet(valuesSubset),
 	}
 
 	_, err := conn.CreateLFTag(input)
 	if err != nil {
 		return fmt.Errorf("error creating Lake Formation LF-Tag: %w", err)
 	}
 
+	// If there are more than 50 values, create them in batches of 50 using UpdateLFTag API
+	for i := 50; i < tagValuesLen; i += lfTagsValuesMaxBatchSize {
```
Comment (theophilechevalier) [inline internal/service/lakeformation/lf_tag.go]: `lfTagsValuesMaxBatchSize` rather than `50`?

File: internal/service/lakeformation/lf_tag.go [hunk 3]
```diff
@@ -185,3 +233,10 @@ func validateLFTagValues() schema.SchemaValidateFunc {
 		validation.StringMatch(regexp.MustCompile(`^([\p{L}\p{Z}\p{N}_.:\*\/=+\-@%]*)$`), ""),
 	)
 }
+
+func Max(x, y int) int {
+	if x > y {
+		return x
+	}
+	return y
+}
```
Comment (theophilechevalier) [inline internal/service/lakeformation/lf_tag.go]: Isn't it defined somewhere else in the provider code?

Comment (frco9) [reply internal/service/lakeformation/lf_tag.go]: It seems not, only in the tests but not in "internal" code

**Pull request merged**